### PR TITLE
Cleanup prefix chip:: from zap templates in which has 'using namespace chip'

### LINKED ISSUE #8658: Cleanup prefix chip:: from all cluster implementations in which has 'using namespace chip'
#### Problem
The chip:: prefix is not needed most of the places it's being used here, since those files are doing using namespace chip. 

 #### Proposed Solution
<suggested fix, suggested enhancement>

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* This is the follow up of #8771, and clean up the prefix 'chip::' in generated code.
* Fixes #8658

#### Change overview
Cleanup prefix chip:: from zap templates files.

#### Testing
How was this tested? (at least one bullet point required)
* No logic change, the regression test is covered by CI.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by mspang — CHANGES_REQUESTED** [review]
Why not just remove the "using namespace chip" declarations

- The samples are not part of chip
- It makes the type declarations inconsistent between .cpp and .h which is rather unfortunate

I don't get why we do this:

using namespace chip;
using namespace app::Clusters;
using namespace System;

How do you even make sense of this? Why even use namespaces?

Maybe we should collapse some namespace down since nobody seems to want to type the long identifiers we have now.

**Comment by mspang** [comment]
To be clear, I'm only objecting to

using namespace chip;

and the other using declarations, which are frankly a bad practice. It can even break your program or make it brittle in the face of library upgrades. It makes namespaces pointless. Experts are in agreement on this - heck, it's on the isocpp.com faq in the context of "using namespace std".

I'm not objecting to removing chip:: inside `namespace chip {}`. There is no issue with that and we should do it.

File: examples/pump-app/pump-common/gen/CHIPClusters.cpp [hunk 1]
```diff
@@ -32,6 +32,8 @@
 
 #include <gen/CHIPClientCallbacks.h>
 
+using namespace chip;
```
Comment (mspang) [inline examples/pump-app/pump-common/gen/CHIPClusters.cpp:35]: Why do you even need it? We open the namespace on line 79, can't you just move that upwards?

**Comment by yufengwangca** [comment]
> Why not just remove these declarations
> 
> * The samples are not part of chip
> * It makes the type declarations inconsistent between .cpp and .h which is rather unfortunate
> 
> I don't get why we do this:
> 
> using namespace chip;
> using namespace app::Clusters;
> using namespace System;
> 
> How do you even make sense of this? Why even use namespaces?
> 
> Maybe we should collapse some namespace down since nobody seems to want to type the long identifiers we have now.

I guess we need to first have discussion and reach agreement on the best coding practice. The bottom line is we should enforce consistence throughout code base. Either make all of them use "chip::" prefix or remove prefix "chip::" all together.

It seems it is an established practice not using "chip::" prefix within our SDK core libraries, the controversial places are src/app, src/platform, and examples.

I would prefer to focus on "chip::" prefix first in this PR and leave "if we should remove namespace" to another PR.

**Comment by mspang** [comment]
> > Why not just remove these declarations
> > 
> > * The samples are not part of chip
> > * It makes the type declarations inconsistent between .cpp and .h which is rather unfortunate
> > 
> > I don't get why we do this:
> > using namespace chip;
> > using namespace app::Clusters;
> > using namespace System;
> > How do you even make sense of this? Why even use namespaces?
> > Maybe we should collapse some namespace down since nobody seems to want to type the long identifiers we have now.
> 
> I guess we need to first have discussion and reach agreement on the best coding practice. The bottom line is we should enforce consistence throughout code base. Either make all of them use "chip::" prefix or remove prefix "chip::" all together.
> 
> It seems it is an established practice not using "chip::" prefix within our SDK core libraries, the controversial places are src/app, src/platform, and examples.

There is no controversy that "chip" is not required within the core library (i.e., for code that is inside "namespace chip {}"). Whether the ZAP generated code is included is maybe a but muddier, but it looks like we do put it in "namespace chip".

Examples are not inside namespace "chip" so I'd say they should be using the chip qualifier and doing so consistently. More generally, they should be written "as if" they were not part of the main repo, because they are there to showcase how to do that. Developers may base their code on those and 

And all of the other namespaces we have are frankly unnecessary. It's clear people don't want to type them - this suggests they should just be removed.

> 
> I would prefer to focus on "chip::" prefix first in this PR and leave "if we should remove namespace" to another PR.

**Comment by mspang** [comment]
> qualifier



> > > Why not just remove these declarations
> > > 
> > > * The samples are not part of chip
> > > * It makes the type declarations inconsistent between .cpp and .h which is rather unfortunate
> > > 
> > > I don't get why we do this:
> > > using namespace chip;
> > > using namespace app::Clusters;
> > > using namespace System;
> > > How do you even make sense of this? Why even use namespaces?
> > > Maybe we should collapse some namespace down since nobody seems to want to type the long identifiers we have now.
> > 
> > 
> > I guess we need to first have discussion and reach agreement on the best coding practice. The bottom line is we should enforce consistence throughout code base. Either make all of them use "chip::" prefix or remove prefix "chip::" all together.
> > It seems it is an established practice not using "chip::" prefix within our SDK core libraries, the controversial places are src/app, src/platform, and examples.
> 
> There is no controversy that "chip" is not required within the core library (i.e., for code that is inside "namespace chip {}"). Whether the ZAP generated code is included is maybe a bit muddier, but it looks like we do put it in "namespace chip".
> 
> Examples are not inside namespace "chip" so I'd say they should be using the chip qualifier and doing so consistently. More generally, they should be written "as if" they were not part of the main repo, because they are there to showcase how to do that. Developers may base their code on those and it would help not lead them into bad practices.
> 
> And all of the other namespaces we have are frankly unnecessary. It's clear people don't want to type them - this suggests they should just be removed.
> 
> > I would prefer to focus on "chip::" prefix first in this PR and leave "if we should remove namespace" to another PR.

It's a good point about DeviceLayer, that one is definitely a bit more ambiguous.

**Comment by mspang** [comment]
> I would prefer to focus on "chip::" prefix first in this PR and leave "if we should remove namespace" to another PR.

See my file comment, I think you might be able to solve my concern just by moving some braces around.

**Comment by yufengwangca** [comment]
> Examples are not inside namespace "chip" so I'd say they should be using the chip qualifier and doing so consistently.

I agree, example code are definitely out side of "chip", src/app contains both ember code  and Interaction Model Protocol implementations, should they are also considered outside of "chip"? I think Interaction Model Protocol  should be treated as same as other protocols even though its implementation is under src/app.

**Comment by bzbarsky-apple** [comment]
`src/app` is code SDK and should `namespace chip { }` as needed.

That said, a bunch of the function names in the ember parts of the SDK can't be easily namespaced yet while conceptually they are part of the core library.  Until we change the names around to themselves be namespaced (which we are in fact planning to do) it might make sense to `using namespace chip` in those core-library implementation files for now.

File: examples/pump-app/pump-common/gen/CHIPClusters.cpp (hunk 1 above)
Comment (yufengwangca) [reply examples/pump-app/pump-common/gen/CHIPClusters.cpp]: acked

**Comment by yufengwangca** [comment]
> Why not just remove the "using namespace chip" declarations
> 
> * The samples are not part of chip
> * It makes the type declarations inconsistent between .cpp and .h which is rather unfortunate
> 
> I don't get why we do this:
> 
> using namespace chip;
> using namespace app::Clusters;
> using namespace System;
> 
> How do you even make sense of this? Why even use namespaces?
> 
> Maybe we should collapse some namespace down since nobody seems to want to type the long identifiers we have now.

This is the summary of our conversation above and please lets me know if there is anything not accurate.
1. "using namespace" is not a recommended practice  and we should avoid introduce new "using namespace" in our code base.
2. All chip core libs including src/app should be placed under namespace chip { }, this need to be addressed in a separate PR.

**Review by mspang — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Comment by yufengwangca** [comment]
@bzbarsky-apple @Damian-Nordic @jepenven-silabs ?

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**